internal/scheduler/testfixtures_test.go

```diff
@@ -0,0 +1,427 @@
+	config.IndexedTaints = append(config.IndexedTaints, indexedTaints...)
+	return config
+}
+
+func withIndexedNodeLabelsConfig(indexedNodeLabels []string, config configuration.SchedulingConfig) configuration.SchedulingConfig {
+	config.IndexedNodeLabels = append(config.IndexedNodeLabels, indexedNodeLabels...)
+	return config
+}
+
+func withUsedResourcesNodes(p int32, rl schedulerobjects.ResourceList, nodes []*schedulerobjects.Node) []*schedulerobjects.Node {
+	for _, node := range nodes {
+		schedulerobjects.AllocatableByPriorityAndResourceType(node.AllocatableByPriorityAndResource).MarkAllocated(p, rl)
+	}
+	return nodes
+}
+
+func withLabelsNodes(labels map[string]string, nodes []*schedulerobjects.Node) []*schedulerobjects.Node {
+	for _, node := range nodes {
+		if node.Labels == nil {
+			node.Labels = maps.Clone(labels)
+		} else {
+			maps.Copy(node.Labels, labels)
+		}
+	}
+	return nodes
+}
+
+func withNodeSelectorPodReqs(selector map[string]string, reqs []*schedulerobjects.PodRequirements) []*schedulerobjects.PodRequirements {
+	for _, req := range reqs {
+		req.NodeSelector = maps.Clone(selector)
+	}
+	return reqs
+}
+
+func withNodeAffinityPodReqs(nodeSelectorTerms []v1.NodeSelectorTerm, reqs []*schedulerobjects.PodRequirements) []*schedulerobjects.PodRequirements {
+	for _, req := range reqs {
+		if req.Affinity == nil {
+			req.Affinity = &v1.Affinity{}
+		}
+		if req.Affinity.NodeAffinity == nil {
+			req.Affinity.NodeAffinity = &v1.NodeAffinity{}
+		}
+		if req.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+			req.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution = &v1.NodeSelector{}
+		}
+		req.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = append(
+			req.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms,
+			nodeSelectorTerms...,
+		)
+	}
+	return reqs
+}
+
+func withGangAnnotationsPodReqs(reqs []*schedulerobjects.PodRequirements) []*schedulerobjects.PodRequirements {
+	gangId := uuid.NewString()
+	gangCardinality := fmt.Sprintf("%d", len(reqs))
+	return withAnnotationsPodReqs(
+		map[string]string{testGangIdAnnotation: gangId, testGangCardinalityAnnotation: gangCardinality},
+		reqs,
+	)
+}
+
+func withAnnotationsPodReqs(annotations map[string]string, reqs []*schedulerobjects.PodRequirements) []*schedulerobjects.PodRequirements {
+	for _, req := range reqs {
+		if req.Annotations == nil {
+			req.Annotations = make(map[string]string)
+		}
+		maps.Copy(req.Annotations, annotations)
+	}
+	return reqs
+}
+
+func withRequestsPodReqs(rl schedulerobjects.ResourceList, reqs []*schedulerobjects.PodRequirements) []*schedulerobjects.PodRequirements {
+	for _, req := range reqs {
+		maps.Copy(
+			req.ResourceRequirements.Requests,
+			schedulerobjects.V1ResourceListFromResourceList(rl),
+		)
+	}
+	return reqs
+}
+
+func testNSmallCpuJob(priority int32, n int) []*schedulerobjects.PodRequirements {
```

Comment:
Can we rename this `testNOneCpuJob`

Reply:
Same for the other methods here.  It also means that someone is less likely to change what "small" means without thinking about it.

Reply:
That'd be good. I'll improve these fixtures as I write more tests.